server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/delete/TransportDeleteSnapshotAction.java

```diff
@@ -60,13 +61,26 @@ protected ClusterBlockException checkBlock(DeleteSnapshotRequest request, Cluste
         return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_READ);
     }
 
+    @Override
+    protected void doExecute(Task task, DeleteSnapshotRequest request, ActionListener<AcknowledgedResponse> listener) {
+        if (clusterService.state().getMinTransportVersion().before(TransportVersions.DELETE_SNAPSHOTS_ASYNC_ADDED)
+            && request.waitForCompletion() == false) {
+            throw new UnsupportedOperationException("wait_for_completion parameter is not supported by all nodes in this cluster");
+        }
+        super.doExecute(task, request, listener);
+    }
+
     @Override
     protected void masterOperation(
         Task task,
         final DeleteSnapshotRequest request,
         ClusterState state,
         final ActionListener<AcknowledgedResponse> listener
     ) {
-        snapshotsService.deleteSnapshots(request, listener.map(v -> AcknowledgedResponse.TRUE));
+        if (request.waitForCompletion()) {
```

Comment:
I wonder if we should push this branch into the `snapshotsService`, seeing as how we pass in the `request` either way. We could conditionally call `addDeleteListener` I think.